### PR TITLE
Backport of #253

### PR DESCRIPTION
**Pull request by fselmo**
### What was wrong?

Backport of #253 into `v0.11.x` branch. All were successful cherry-pick commits without conflicts.

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-account/blob/main/newsfragments/README.md)

#### Cute Animal Picture

Screenshot 2024-03-19 at 09 53 18

### CODE REVIEW

File: eth_account/_utils/typed_transactions/base.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                        compute_blob_kzg_proof(
+                            blob.data,
+                            commitment.data,
+                            load_trusted_setup(TRUSTED_SETUP),
+                        )
+                    )
+                )
+                for blob, commitment in zip(self.blobs, self.commitments)
+            ]
+        return self._proofs
```
Comment (kclowes) [inline eth_account/_utils/typed_transactions/base.py]: Feel free to take or leave, but maybe this section warrants its own file?

File: tests/core/_test_data/blob_data_1.txt
Comment (kclowes) [inline tests/core/_test_data/blob_data_1.txt:1]: Lazy question - did not check the tests... Do we need to add these `txt` files to the Manifest too?

**Review by kclowes — APPROVED** [review]
I left some take or leave comments, but looks good!

File: eth_account/_utils/typed_transactions/base.py (hunk 1 above)
Comment (fselmo) [reply eth_account/_utils/typed_transactions/base.py]: 💯 I do like that

File: tests/core/_test_data/blob_data_1.txt
Comment (fselmo) [reply tests/core/_test_data/blob_data_1.txt:1]: I sanity checked the `/dist` and everything in `tests/` looks to be included, including these test data files, so I think we're good here.

**Pull request merged**